README.md

```diff
@@ -54,6 +54,113 @@ The following Visual Studio Code extensions are recommended:
 - [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
 - [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
 
+## Scripts and tasks
+
+Scripts for each `package.json` should follow these rules:
+
+### Nesting
+
+Some scripts may be nested under other scripts using the colon (`:`) symbol. This can interchangeably be used to mean either:
+
+- The nested script is a fragment of the parent script.
+  - E.g.:
+    ```json
+    {
+      "build": "npm run build:bundles && npm run build:definitions",
+      "build:bundles": "...",
+      "build:definitions": "..."
+    }
+    ```
+- The nested script is an alternative configuration of the parent script.
+  - E.g.:
+    ```json
+    {
+      "e2e": "cypress run --browser chrome",
+      "e2e:firefox": "cypress run --browser firefox",
+      "e2e:watch": "cypress open"
+    }
+    ```
+
+### `build` script
+
+- Generates all the files needed for a dependent project to run.
+- Doesn't expect another script from the same project to be run before this script.
+- If the build fails, exits with an error.
+
+### `dev` script
```

Comment:
The choice to use `dev` instead of `start` or `serve` is very debatable, I'd be curious what your opinions are.

I chose `dev` because it's unambiguous.
  * `start` could mean "start the current (production) build".
  * `serve` could mean "serve the current (production) build".

I think `start` would also be a great choice since it's one of NPM's defaults (we can run it with `npm start` instead of `npm run start`). However, it's a little bit more ambiguous.

Reply:
https://github.com/coveo/ui-kit/pull/2724/files#r1106304160 related